### PR TITLE
Zabbix: Use Agent2 as Default | Update Script added | some other Improvements

### LINKED ISSUE #174: Zabbix LXC script installs Zabbix agent (1), not Zabbix agent 2, which results in an alert for high memory utilization for LXC containers including Zabbix itself
### Please verify that you have read and understood the guidelines.

Yes

### A clear and concise description of the issue.

When using https://community-scripts.github.io/ProxmoxVE/scripts?id=zabbix the installation comes with Zabbix agent 1, which does not handle LXC memory utilization correctly.

This can be fixed by modifying Zabbix parameters and template or by installing Zabbix agent 2:
https://github.com/kvaps/zabbix-linux-container-template

I fixed the issues for myself by uninstalling Zabbix agent (1) from the Zabbix server and installing Zabbix agent 2.

Could the script be changed so it uses Zabbix Agent 2 by default?

### What settings are you currently utilizing?

Default Settings

### Which Linux distribution are you employing?

Debian 12

### If relevant, including screenshots or a code block can be helpful in clarifying the issue.

_No response_

### Please provide detailed steps to reproduce the issue.

Install Zabbix LXC using default settings, log into Zabbix.

### PR DESCRIPTION
**Pull request by MickLesk**
## Description
- Updating the code base
- Script flow adapted to standard
- Switch to Zabbix-Agent2 (for LXC's)
- Update script with backup data
- some other little improvements

Breaking Change, since the script is now running with a different agent, I cannot and do not want to anchor this in the CT.sh script, it causes too many problems. So if you want to use Agent2, you have to install it yourself.

Fixes #174 

## Type of change
Please check the relevant option(s):

- [x] Bug fix (non-breaking change that resolves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [x] Breaking change (a fix or feature that would cause existing functionality to change unexpectedly)
- [ ] New script (a fully functional and thoroughly tested script or set of scripts.)

## Prerequisites
The following efforts must be made for the PR to be considered. Please check when completed:
- [x] Self-review performed (I have reviewed my code, ensuring it follows established patterns and conventions)
- [x] Testing performed (I have tested my changes, ensuring everything works as expected)
- [x] Documentation updated (I have updated any relevant documentation)

### CODE REVIEW

File: ct/zabbix.sh [hunk 1]
```diff
@@ -57,11 +57,31 @@ header_info
...
+msg_ok "Stopped ${APP} Services"
+
 msg_info "Updating $APP LXC"
+mkdir -p /opt/zabbix-backup/
+cp /etc/zabbix/zabbix_server.conf /opt/zabbix-backup/
+cp /etc/apache2/conf-enabled/zabbix.conf /opt/zabbix-backup/
+cp -R /usr/share/zabbix/ /opt/zabbix-backup/
+cp -R /usr/share/zabbix-* /opt/zabbix-backup/
+cd /tmp
+wget -q https://repo.zabbix.com/zabbix/7.0/debian/pool/main/z/zabbix-release/zabbix-release_latest+debian12_all.deb
```
Comment (havardthom) [inline ct/zabbix.sh:71]: https://www.zabbix.com/documentation/current/en/manual/installation/upgrade/packages/debian_ubuntu#update-repository-configuration-package

`rm -Rf /etc/apt/sources.list.d/zabbix.list` is this step not needed?

File: json/zabbix.json [hunk 2]
```diff
@@ -33,7 +33,11 @@
     "notes": [
         {
             "text": "Database credentials: `cat zabbix.creds`",
-            "type": "warning"
+            "type": "info"
+        }
+        {
+            "text": "Zabbix is running in these LXC as standard with Zabbix-Agent2",
```
Comment (havardthom) [inline json/zabbix.json:39]:
```suggestion
            "text": "Zabbix agent 2 is used by default",
```

**Review by havardthom — DISMISSED** [review]
Nice 👍

File: ct/zabbix.sh (hunk 1 above)
Comment (MickLesk) [reply ct/zabbix.sh:72]: I have included it as an option, it probably shouldn't matter. At least it worked without and with the same. Presumably this is there for major updates, e.g. from 7.0 to 8.0 or something like that.

**Review by havardthom — APPROVED** [review]

**Pull request merged**